ios/Classes/StripePaymentPlugin.m

```diff
@@ -21,8 +21,140 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+                FlutterError* flutterError = [FlutterError errorWithCode:errorCode message:error.localizedFailureReason details:nil];
+                result(flutterError);
+            } else {
+                result(token);
+            }
+        }];
+        
+    } else if ([@"createSourceWithParams" isEqualToString:call.method]) {
+        
+        NSDictionary* argDict = [NSDictionary dictionaryWithDictionary:call.arguments];
+        NSNumber* amount = [argDict objectForKey:@"amount"];
+        NSString* currency = [argDict objectForKey:@"currency"];
+        NSString* returnURL = [argDict objectForKey:@"returnURL"];
+        NSString* type = [argDict objectForKey:@"type"];
+        
+        STPSourceParams* params = [[STPSourceParams alloc] init];
+        params.amount = amount;
+        params.currency = currency;
+        
+        STPSourceType sourceType = STPSourceTypeUnknown;
+        if ([type localizedCaseInsensitiveCompare:@"eps"]) {
+            sourceType = STPSourceTypeEPS;
+        } else if ([type localizedCaseInsensitiveCompare:@"p24"]) {
+            sourceType = STPSourceTypeP24;
+        } else if ([type localizedCaseInsensitiveCompare:@"card"]) {
+            sourceType = STPSourceTypeCard;
+        } else if ([type localizedCaseInsensitiveCompare:@"ideal"]) {
+            sourceType = STPSourceTypeIDEAL;
+        } else if ([type localizedCaseInsensitiveCompare:@"alipay"]) {
+            sourceType = STPSourceTypeAlipay;
+        } else if ([type localizedCaseInsensitiveCompare:@"wechatpay"]) {
+            sourceType = STPSourceTypeWeChatPay;
+        } else if ([type localizedCaseInsensitiveCompare:@"giropay"]) {
+            sourceType = STPSourceTypeGiropay;
+        } else if ([type localizedCaseInsensitiveCompare:@"sepadebit"]) {
+            sourceType = STPSourceTypeSEPADebit;
+        } else if ([type localizedCaseInsensitiveCompare:@"bancontact"]) {
+            sourceType = STPSourceTypeBancontact;
+        } else if ([type localizedCaseInsensitiveCompare:@"sofort"]) {
+            sourceType = STPSourceTypeSofort;
+        } else if ([type localizedCaseInsensitiveCompare:@"multibanco"]) {
+            sourceType = STPSourceTypeMultibanco;
+        } else if ([type localizedCaseInsensitiveCompare:@"threedsecure"]) {
+            sourceType = STPSourceTypeThreeDSecure;
+        }
+        
+        params.type = sourceType;
+        
+        params.redirect = @{@"flow": @"redirect", @"returnURL": returnURL};
+        
+        [STPAPIClient.sharedClient createSourceWithParams:params completion:^(STPSource * _Nullable source, NSError * _Nullable error) {
+            if (error) {
+                NSString* errorCode = [NSString stringWithFormat:@"%li", error.code];
+                FlutterError* flutterError = [FlutterError errorWithCode:errorCode message:error.localizedFailureReason details:nil];
+                result(flutterError);
+            } else {
```

Comment:
why not stick to the exact same source code as in https://github.com/tipsi/tipsi-stripe/blob/master/ios/TPSStripe/TPSStripeManager.m#L259? 
To minimize problems and to keep our source code up to date with theirs we should try to keep differences as minimal as possible

Reply:
Oh yes this looks like the right way to do it, I’ll change this, thank you!